### ISSUE TITLE
Compiled 'Master Branch' - Nicotine is Black Blank Screen?

### ISSUE THREAD

**Issue by 0pLuS0** [Nicotine-Plus/nicotine-plus#140]
Hello,

I'm running Slackware 14.2 x86_64, not sure if things here are to old, but this what I have.

Nicotine - Master Branch
gobject-introspection-1.46.0
mutagen-1.44.0
pygobject-3.28.3

gir1.2-gtk-3.0 - I don't know what this is?

Slack 14.2 uses gtk+3-3.18.9

When I run nicotine from a term, this is the output;

~ >nicotine 
Tue 21:17 Nicotine+ supports a country code blocker. This requires a (GPL'ed) library called GeoIP. You can find it here: C library:       http://www.maxmind.com/app/c Python
          bindings: http://www.maxmind.com/app/python (the python bindings require the C library)
Traceback (most recent call last):
  File "/usr/bin/nicotine", line 276, in <module>
    run()
  File "/usr/bin/nicotine", line 252, in run
    app = frame.MainApp(data_dir, config, plugins, trayicon, hidden, bindip, port)
  File "/usr/lib64/python3.7/site-packages/pynicotine/gtkgui/frame.py", line 3730, in __init__
    self.frame = NicotineFrame(data_dir, config, plugins, trayicon, start_hidden, bindip, port)
  File "/usr/lib64/python3.7/site-packages/pynicotine/gtkgui/frame.py", line 454, in __init__
    self.LogScrolledWindow.set_propagate_natural_height(True)
AttributeError: 'ScrolledWindow' object has no attribute 'set_propagate_natural_height'
zsh: terminated  nicotine

```THANKS```

**Comment by kiplingw** [comment]
@0pLuS0, I appreciate that you are not using a Debian based distro. But unless you are a developer, please avoid running from source. Instead consider trying the latest unstable from the PPA. This gives us confidence it was installed correctly on your machine with the appropriate dependencies before you attempted to run it.

**Comment by mathiascode** [comment]
Seems like the 'set_propagate_natural_height' attribute was added in GTK 3.22. I'll see if we can work around this in older GTK versions.

**Comment by 0pLuS0** [comment]
@kiplingw 

With all due respect, you are the first person to say something like this to me in 20 years of running, compiling software and building slackware packages in Slackware. I don't know everything, but who does? 

I've been doing System Admin, IT, Security, Kernel and application compiling and package building in Slackware for 20 years. I don't know everything, but who does, and no one needs to be a developer to compile code.

I've seen for the past 20 years that Ubuntu and Debian will many times have their own naming schemes on source that other distros don't follow/use.

As example it's mentioned here for Nicotine;

python3-gi, but that is typically listed from what I've seen as gobject-introspection

gir1.2-gtk-3.0 - is my understanding, suppose to be the gir bindings in GTK3.

From your list of DEPS I have these;

python3-3.7.2
pygobject3-python3-3.18.2
mutagen-1.44.0
gtk+3-3.18.9

So if I can understand better python3-gi, and gir1.2-gtk-3.0, then I'll figure this out, but probably given the age of Slack 14.2, if it needs anything in the latest versions of GTK3, then I know I will have a problem here, but I also thought Nicotine is using Python, and it doesn't matter the GTK version, at least I thought so with 1.4.1

By the way, this was my last Slackware Build Script I made for 1.,4.1


#!/bin/sh
#
# Slackware build script for nicotine-plus

PRGNAM=nicotine-plus
VERSION=${VERSION:-1.4.1}
BUILD=${BUILD:-2}
TAG=${TAG:-_sar}

if [ -z "$ARCH" ]; then
  case "$( uname -m )" in
    i?86) ARCH=i486 ;;
    arm*) ARCH=arm ;;
       *) ARCH=$( uname -m ) ;;
  esac
fi

CWD=$(pwd)
TMP=${TMP:-/tmp/sar}
PKG=$TMP/package-$PRGNAM
OUTPUT=${OUTPUT:-/tmp}

set -eu

rm -rf $PKG
mkdir -p $TMP $PKG $OUTPUT
cd $TMP
rm -rf $PRGNAM-$VERSION
tar xvf $CWD/$PRGNAM-$VERSION.tar.?z*
cd $PRGNAM-$VERSION
chown -R root:root .
find -L . \
 \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
  -o -perm 511 \) -exec chmod 755 {} \; -o \
 \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
 -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;

python3 setup.py install --root=$PKG

mv $PKG/usr/share/man $PKG/usr

find $PKG/usr/man -type f -exec gzip -9 {} \;
for i in $( find $PKG/usr/man -type l ) ; do ln -s $( readlink $i ).gz $i.gz ; rm $i ; done

rm -rf $PKG/usr/share/locale/

mkdir -p $PKG/install
cat $CWD/slack-desc > $PKG/install/slack-desc
cat $CWD/doinst.sh > $PKG/install/doinst.sh

cd $PKG
/sbin/makepkg -l y -c n $OUTPUT/$PRGNAM-$VERSION-$ARCH-$BUILD$TAG.${PKGTYPE:-tgz}

**Comment by mathiascode** [comment]
@0pLuS0 Can you test this branch? https://github.com/mathiascode/nicotine-plus/tree/chatfixes

**Comment by zniavre** [comment]
https://pastebin.com/j75fUBY5

 @0pLuS0 if you do not mind i tried ...

**Comment by 0pLuS0** [comment]
@mathiascode 

What about the DEPS I listed above?

python3-3.7.2
pygobject3-python3-3.18.2
mutagen-1.44.0
gtk+3-3.18.9

THANKS

**Comment by mathiascode** [comment]
@0pLuS0 Are you sure you pulled the correct branch? There's no "set_propagate_natural_height" attribute in my updated code.

@zniavre Can you update the branch and test again? I think I've solved the log issue.

**Comment by zniavre** [comment]
@mathiascode  >  https://pastebin.com/yncF5AGT

**Comment by mathiascode** [comment]
@zniavre Looks fine. Are there any crashes anymore? Do chats work fine?

I'm having trouble reproducing the Alt+R issue you reported earlier. What's your GTK version, and are you using any custom GTK themes?

**Comment by 0pLuS0** [comment]
@mathiascode 

```git clone https://github.com/mathiascode/nicotine-plus.git```

**Comment by mathiascode** [comment]
Do `git clone -b chatfixes https://github.com/mathiascode/nicotine-plus.git`

**Comment by zniavre** [comment]
libgtk-3-0:amd64  3.24.12-1ubuntu1 amd64
on Ubuntu 19.10 (minimal installation running Unity )
my own gtk+3.x theme (it run without errors ...)
got same gdk warnings with ubuntu default theme

**Comment by 0pLuS0** [comment]
@mathiascode 

Ok here it is, oh a thing of Beauty, and not one error in the terminal either when I started it! WoOT...

Now who was it that said if I wasn't a Developer? LOL... Just kidding... :)

So was the issue with the original GTK?

![nicotine](https://user-images.githubusercontent.com/36190223/82435961-971f6880-9a84-11ea-9c08-6940dbe4c0b2.jpg)


```THANKS```

```P.S. I'll post bug reports later on your Git, I found a UI bug```

**Comment by mathiascode** [comment]
The issue was the older GTK version that didn't include a newer feature. Glad to see it's running at least. :)

Feel free to create new bug reports. The Python3/GTK3 port hasn't really been tested on GTK 3.18 yet, so there could be more breaking issues.

**Comment by 0pLuS0** [comment]
Ok, I assumed it was GTK related, hopefully the issue I saw is not an older version issue, but something helpful for you...

Here she is, and oh my, it's been like 2 years waiting on development to run nicotine again...

![Nicotine](https://user-images.githubusercontent.com/36190223/82436759-dc906580-9a85-11ea-841c-73804631263e.jpg)


```THANKS```

**Comment by zniavre** [comment]
if statusbar is oversized it's caused by the progressbar (from scanning) position... but i do not know what to do   :o)

**Comment by mathiascode** [comment]
Oversized statusbar now fixed in https://github.com/Nicotine-Plus/nicotine-plus/pull/141

**Comment by zniavre** [comment]
@mathiascode looks better for me but...

https://imgur.com/a/1NLYHa4

statusbar from few days ago / gtk2 vs gtk3 / and today with your last commit, wich is really good ! 
 no more gtk/gdk warnings 
thank you all to work on nicotine !

**Comment by mathiascode** [comment]
It's going to be slightly taller than the GTK2 statusbar in any case, due to GTK3 defaults, but I'll see if hiding the progressbar text makes the statusbar less tall with your theme.

Thanks!

**Comment by mathiascode** [comment]
I just pushed a change to the https://github.com/Nicotine-Plus/nicotine-plus/pull/141 PR. Can you try again?

**Comment by zniavre** [comment]
i was reactive...
 https://imgur.com/a/M1jalhq 
orange is default ubuntu 
black is mine

**Comment by mathiascode** [comment]
That seems to be the correct size.

**Comment by zniavre** [comment]
https://imgur.com/a/4zKenGK

look at the statusbar of another pure gtk+3

**Comment by mathiascode** [comment]
Noted, but it's not a priority for me right now. The whole UI needs a good cleanup sometime in the future, since there are a lot of deprecated elements.
